cupy/core/_routines_statistics.pyx

```diff
@@ -338,6 +348,52 @@ cdef _mean = create_reduction_func(
      'out0 = a / _type_reduce(_in_ind.size() / _out_ind.size())', None))
 
 
+_count_non_nan = create_reduction_func(
+    'cupy_count_non_nan',
+    ('e->e', 'f->f', 'd->d'),
+    ('(in0 == in0) ? 1.0 : 0.0', 'a + b', 'out0 = a', None), 0)
+
+
+cdef ndarray _nanstd(
+        ndarray a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+    ret = _nanvar(
+        a, axis=axis, dtype=dtype, out=None, ddof=ddof, keepdims=keepdims)
+    return _math._sqrt(ret, dtype=dtype, out=out)
+
+
+cdef ndarray _nanvar(
+        ndarray a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+
+    assert a.dtype.kind != 'c', 'Variance for complex numbers is not ' \
+                                'implemented. Current implemention does not ' \
+                                'convert the dtype'
+
+    _count = _count_non_nan(a, axis=axis, dtype=dtype, keepdims=True)
+    arrmean = a._nansum(axis=axis, dtype=dtype, out=None, keepdims=True)
+    arrmean = arrmean / _count
+
+    alpha = _count - ddof
+    alpha = alpha.clip(a_min=0, a_max=None)
+    alpha = 1./alpha
```

Comment:
How about writing them in `ReductionKenrel` to reduce the number of kernel launch?

Reply:
Done